rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeJavaCompatibility.java

```diff
@@ -0,0 +1,178 @@
+package org.openrewrite.gradle;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.gradle.util.ChangeStringLiteral;
+import org.openrewrite.groovy.GroovyVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class ChangeJavaCompatibility extends Recipe {
+    @Option(displayName = "New version",
+            description = "The Java version to update source compatibility to",
```

Comment:
Consider mentioning the valid values. If someone wants Java 8 should they put in "8" or "1.8"?